tests/core/io/test_udp_server.h

```diff
@@ -0,0 +1,283 @@
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_UDP_SERVER_H
+#define TEST_UDP_SERVER_H
+
+#include "core/io/packet_peer_udp.h"
+#include "core/io/udp_server.h"
+#include "tests/test_macros.h"
+
+namespace TestUDPServer {
+
+const int port = 12345;
+const IPAddress localhost("127.0.0.1");
+
+Ref<UDPServer> create_server(const IPAddress &p_address, int p_port) {
+	Ref<UDPServer> server;
+	server.instantiate();
+
+	Error err = server->listen(port, localhost);
+	REQUIRE_EQ(Error::OK, err);
+	REQUIRE(server->is_listening());
+	CHECK_FALSE(server->is_connection_available());
+	CHECK_EQ(16, server->get_max_pending_connections());
+
+	return server;
+}
+
+Ref<PacketPeerUDP> create_client(const IPAddress &p_address, int p_port) {
+	Ref<PacketPeerUDP> client;
+	client.instantiate();
+
+	Error err = client->connect_to_host(localhost, port);
+	REQUIRE_EQ(Error::OK, err);
+	CHECK(client->is_bound());
+	CHECK(client->is_socket_connected());
+
+	return client;
+}
+
+Ref<PacketPeerUDP> accept_connection(Ref<UDPServer> &server) {
+	Error err = server->poll();
+	CHECK_EQ(Error::OK, err);
+
+	REQUIRE(server->is_connection_available());
+	Ref<PacketPeerUDP> client_from_server = server->take_connection();
+	REQUIRE_MESSAGE(client_from_server.is_valid(), "A Packet Peer UDP from the UDP Server should not be a null pointer");
+	CHECK(client_from_server->is_bound());
+	CHECK(client_from_server->is_socket_connected());
+
+	return client_from_server;
+}
+
+TEST_CASE("[UDPServer] Instantiation") {
+	Ref<UDPServer> server;
+	server.instantiate();
+
+	REQUIRE_MESSAGE(server.is_valid(), "A UDP Server created should not be a null pointer");
+	CHECK_EQ(server->is_listening(), false);
+}
+
+TEST_CASE("[UDPServer] Accept a connection and receive/send data") {
+	Ref<UDPServer> server = create_server(localhost, port);
+	Ref<PacketPeerUDP> client = create_client(localhost, port);
+
+	// Sending data from client to server
+	String hello_world = "Hello World!";
+	CHECK_EQ(Error::OK, client->put_var(hello_world));
+
+	// Required to get the connection properly stablished
```

Comment:
```suggestion
	// Required to get the connection properly established
```

Reply:
```suggestion
	// Required to get the connection properly established.
```
Period at end as well